client.lua

```diff
@@ -922,26 +924,38 @@ local function IsWhitelistedWeaponStress(weapon)
     return false
 end
 
-local shootingSleep = 500
-CreateThread(function() -- Shooting
-    while true do
-        local isArmed = IsPedArmed(cache.ped, 7)
-        if LocalPlayer.state.isLoggedIn and isArmed then
-            local ped = cache.ped
-            local weapon = GetSelectedPedWeapon(ped)
-            if weapon ~= `WEAPON_UNARMED` then
-                if IsPedShooting(ped) and not IsWhitelistedWeaponStress(weapon) then
-                    if math.random() < config.StressChance then
-                        TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
-                    end
+local function startWeaponStressThread(weapon)
+    if IsWhitelistedWeaponStress(weapon) then return end
+    hasWeapon = true
+
+    CreateThread(function()
+        while hasWeapon do
+            if IsPedShooting(cache.ped) then
+                if math.random() <= config.StressChance then
+                    TriggerServerEvent('hud:server:GainStress', math.random(1, 3))
                 end
-                shootingSleep = 0
-            else
-                shootingSleep = 1000
             end
+            Wait(0)
         end
-        Wait(shootingSleep)
+    end)
+end
+
+local function getCurrentWeapon()
+    if LocalPlayer.state.isLoggedIn then
+        local weapon = OxInventory:getCurrentWeapon()
+        if weapon then startWeaponStressThread(weapon.name) end
     end
+end
+
+getCurrentWeapon()
```

Comment:
Seems like this might need to be put inside of a player loaded event handler rather than being at init time?

Reply:
According to what David and Berkie discussed here, it's better that we go this way since it's the client executing it and the server should already have loaded: https://github.com/Qbox-project/qbx-hud/pull/4#discussion_r1183123040

Reply:
I'm not convinced by that comment thread, but putting that aside for a moment, is this even needed? As far as I'm aware the ox_inventory event will also get triggered since players don't spawn with guns in hand correct?

Reply:
Tested and you are right, if you leave a weapon in hand and relog, it'll appear without a gun, so no need to run this thread at the beginning